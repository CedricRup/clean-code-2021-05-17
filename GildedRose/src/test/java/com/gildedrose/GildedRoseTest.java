package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    private static final String SULFURAS = "Sulfuras, Hand of Ragnaros";

    @Test
    void should_decrease_item_quality() {
        // Given
        String[] names = new String[]{ "Regular Item", SULFURAS };
        Integer[] sellIns = new Integer[]{ 3, 0 };
        Integer[] qualities = new Integer[]{ 10, 0 };

        // When / Then
        CombinationApprovals.verifyAllCombinations(this::getItem, names, sellIns, qualities);
    }

    private Item getItem(String name, int sellIn, int quality) {
        Item regularItem = new Item(name, sellIn, quality);

        GildedRose gildedRose = new GildedRose(new Item[]{ regularItem });
        gildedRose.updateQuality();
        return regularItem;
    }

    @Test
    void should_decrease_twice_item_quality() {
        Item regularItem = new Item("Regular Item", 1, 10);
        GildedRose gildedRose = new GildedRose(new Item[]{ regularItem });

        gildedRose.updateQuality();
        gildedRose.updateQuality();

        Assertions.assertEquals(7, regularItem.quality);
    }

    @Test
    void should_decrease_to_zero_item_quality() {
        Item regularItem = new Item("Regular Item", 1, 1);
        GildedRose gildedRose = new GildedRose(new Item[]{ regularItem });

        gildedRose.updateQuality();
        gildedRose.updateQuality();

        Assertions.assertEquals(0, regularItem.quality);
    }

    @Test
    void should_decrease_sulfuras_item_quality() {
        Item regularItem = new Item(SULFURAS, 1, 10);
        GildedRose gildedRose = new GildedRose(new Item[]{ regularItem });

        gildedRose.updateQuality();
        gildedRose.updateQuality();

        Assertions.assertEquals(10, regularItem.quality);
    }

}
