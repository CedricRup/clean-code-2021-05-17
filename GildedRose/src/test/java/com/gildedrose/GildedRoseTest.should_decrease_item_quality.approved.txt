[Regular Item, 3, 10] => Regular Item, 2, 9 
[Regular Item, 3, 0] => Regular Item, 2, 0 
[Regular Item, 0, 10] => Regular Item, -1, 8 
[Regular Item, 0, 0] => Regular Item, -1, 0 
[Sulfuras, Hand of Ragnaros, 3, 10] => Sulfuras, Hand of Ragnaros, 3, 10 
[Sulfuras, Hand of Ragnaros, 3, 0] => Sulfuras, Hand of Ragnaros, 3, 0 
[Sulfuras, Hand of Ragnaros, 0, 10] => Sulfuras, Hand of Ragnaros, 0, 10 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
